src/main/java/com/aws/greengrass/clientdevices/auth/configuration/RuntimeConfiguration.java

```diff
@@ -97,34 +99,38 @@ public void updateCAPassphrase(String passphrase) {
      * @param thingName ThingName
      * @return Optional of ThingV1 DTO, else empty optional
      */
-    public Optional<ThingV1> getThingV1(String thingName) {
+    public Optional<ThingV1DTO> getThingV1(String thingName) {
         Topics v1ThingTopics = config.findTopics(THINGS_KEY, THINGS_V1_KEY, thingName);
 
         if (v1ThingTopics == null) {
             return Optional.empty();
         }
 
+        // defaults to 0 for invalid config values
+        int thingVersion = Coerce.toInt(v1ThingTopics.find(THINGS_VERSION_KEY));
+
         Topics certTopics = v1ThingTopics.findTopics(THINGS_CERTIFICATES_KEY);
         if (certTopics == null || certTopics.isEmpty()) {
-            return Optional.of(new ThingV1(thingName, Collections.emptyMap()));
+            return Optional.of(new ThingV1DTO(thingVersion, thingName, Collections.emptyMap()));
         }
         Map<String, Long> certMap = new HashMap<>();
         certTopics.forEach(node -> {
             certMap.put(node.getName(), Coerce.toLong(node));
         });
 
-        return Optional.of(new ThingV1(thingName, certMap));
+        return Optional.of(new ThingV1DTO(thingVersion, thingName, certMap));
```

Comment:
Why do we still want to pass the thing version here if we are saying it is version1? We are not passing the version for the `CertificateV1DTO`